lib/components/artworks/index.js

```diff
@@ -0,0 +1,224 @@
+    fetch(metaphysicsURL, { method: 'post',
+                            headers: { 'Content-Type': 'application/json' },
+                            body: JSON.stringify({ query }) })
+      .then((response) => {
+        if (response.status >= 200 && response.status < 300) {
+          return response
+        } else {
+          const error = new Error(response.statusText)
+          error.response = response
+          throw error
+        }
+      })
+      .then((response) => response.json())
+      .then(({ data }) => {
+        const artworks = data.artist.artworks;
+        this.setState({ page: nextPage, artworks: this.state.artworks.concat(artworks) });
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }
+
+  onLayout(event: LayoutEvent) {
+    const layout = event.nativeEvent.layout;
+    if (layout.width > 0) {
+      // This is the sum of all margins in between sections, so do not count to the right of last column.
+      const sectionMargins = this.props.sectionMargin * (this.props.sectionCount - 1);
+      this.setState({ sectionDimension: (layout.width - sectionMargins) / this.props.sectionCount });
+    }
+  }
+
+  sectionedArtworks() {
+    const sectionedArtworks = [];
+    const sectionRatioSums = [];
+    for (let i = 0; i < this.props.sectionCount; i++) {
+      sectionedArtworks.push([]);
+      sectionRatioSums.push(0);
+    };
+
+    const artworks = this.state.artworks;
+    for (let i = 0; i < artworks.length; i++) {
+      const artwork = this.state.artworks[i];
+
+      // Find section with lowest *inverted* aspect ratio sum, which is the shortest column.
+      let lowestRatioSum = Number.MAX_VALUE; // Start higher, so we always find a
+      let sectionIndex = null;
+      for (let j = 0; j < sectionRatioSums.length; j++) {
+        const ratioSum = sectionRatioSums[j];
+        if (ratioSum < lowestRatioSum) {
+          sectionIndex = j;
+          lowestRatioSum = ratioSum;
+        }
+      }
+
+      const section = sectionedArtworks[sectionIndex];
+      section.push(artwork);
+
+      // Keep track of total section aspect ratio
+      const aspectRatio = artwork.image.aspect_ratio || 1; // Ensure we never divide by null/0
+      // Invert the aspect ratio so that a lower value means a shorter section.
+      sectionRatioSums[sectionIndex] += (1 / aspectRatio);
+    }
+
+    return sectionedArtworks;
+  }
+
+  renderSections() {
+    const spacerStyle = {
+      height: this.props.itemMargin,
+    };
+
+    const sectionedArtworks = this.sectionedArtworks();
+    const sections = [];
+    for (let i = 0; i < this.props.sectionCount; i++) {
+      const artworkComponents = [];
+      const artworks = sectionedArtworks[i];
+      for (let j = 0; j < artworks.length; j++) {
+        artworkComponents.push(<Artwork artwork={artworks[j]} key={'artwork-'+j} />);
+        // Setting a marginBottom on the artwork component didn’t work, so using a spacer view instead.
```

Comment:
Could definitely have been my ignorance, feel free to take a stab at some point.